ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/AuthenticationUtil.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.util;
+
+import io.confluent.ksql.security.KsqlAuthTokenProvider;
+import io.confluent.ksql.security.KsqlPrincipal;
+import io.confluent.ksql.util.KsqlConfig;
+import java.time.Clock;
+import java.util.Objects;
+import java.util.Optional;
+
+public class AuthenticationUtil {
+
+  final Clock clock;
+
+  public AuthenticationUtil(final Clock clock) {
+    this.clock = Objects.requireNonNull(clock);
+  }
+
+  public Optional<Long> getTokenTimeout(
+      final Optional<KsqlPrincipal> principal,
```

Comment:
Would it make sense to extend the `KsqlPrincipal` interface to provide a getTimeout method?
We could return a default value based on configuration and have the other classes like JWT override the method.

Reply:
That wouldn't work, because the `KsqlPrincipal` is implemented in the ksql library, and extracting timestamps requires a dependency to the security plugins (`JwtPrincipal` does not override `KsqlPrincipal`)